cmd/generate/config/utils/validate.go

```diff
@@ -81,5 +60,29 @@ func ValidateWithPaths(r config.Rule, truePositives map[string]string, falsePosi
 				Msg("Failed to Validate. False positive was detected by regex and/or path.")
 		}
 	}
-	return &r
+	return r
+}
+
+func createDetector(r *config.Rule) *detect.Detector {
```

Comment:
**Suggestion**: @rgmz wdyt about renaming this `createSingleRuleDetector`? At first when I read this code I was like "wait why are we building a detector here?"